rhmap-test-driver/app/src/test-runner.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+const API = require("./rhmap-api");
+
+class TestRunner {
+
+    constructor() {
+        // The ID of the application that owns the target aliases.
+        this.appId;
+        // The time between one request and another
+        this.delay;
+    }
+
+    /**
+     * The URL of the node backend
+     */
+    set endPoint(endPoint) {
+        if (endPoint.lastIndexOf('/') === endPoint.length - 1) {
+            endPoint.slice(-1);
+        }
+        this.api = new API(endPoint);
+    }
+
+    /**
+     * It sends a request to the backend endpoint for each alias with a fix delay between each one
+     */
+    start(aliases) {
+        if (!this.api || !this.appId) {
+            throw new Error("An endpoint URL and an application ID must be provided");
+        }
+
+        const message = "Hello from rhmap-test-driver!";
+
+        aliases.forEach((alias, i) => {
+            setTimeout(() => {
+                console.log(`Sending notification to ${alias}`);
+                this.api.sendNotificationToAlias(message, this.appId, alias)
+                    .on("error", err => {
+                        console.log(`[${alias}] FAILED: ${err}`)
+                    })
+                    .on("response", res => {
+                        console.log(`[${alias}] SUCCESS`)
+                    });;
+            }, this.delay * i);
```

Comment:
I have made a little investigation. Using Async library works well and may be a bit cleaner. However, the intervals *are not* fixed. There is an error of ~6 ms that is accumulating on each iteration:
Current version:
Sending notification to josemi-app at 2
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 5005
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 10004
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 15004
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 20005
[josemi-app] RESPONSE: 202

Async version:
Sending notification to josemi-app at 0
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 5007
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 10032
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 15044
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 20056
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 25069
[josemi-app] RESPONSE: 202

Async version with compensation:
Sending notification to josemi-app at 0
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 5004
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 10009
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 15015
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 20020
[josemi-app] RESPONSE: 202
Sending notification to josemi-app at 25027
[josemi-app] RESPONSE: 202

This is only with 5 request, after 1000 request the error would be 6 seconds!